src/lib/swapper/swappers/ThorchainSwapper/utils/addAggregatorAndDestinationToMemo.ts

```diff
@@ -1,100 +1,98 @@
-  // We trim 10 of precisions to make sure the THORChain parser can handle the amount without precisions and rounding issues
-  // If the finalAssetPrecision is under 5, the THORChain parser won't fail and we add one exponent at the end so the aggregator contract won't multiply the amount
-  const finalAssetLimitWithTwoLastNumbersAsExponent = `${
-    finalAssetPrecision < maximumPrecision
-      ? finalAssetLimitWithManualSlippage
-      : finalAssetLimitCryptoPrecision.replace('.', '')
-  }${thorAggregatorExponential}`
-
-  // Paranoia assertion - expectedAmountOut should never be 0 as it would likely lead to a loss of funds.
-  assert(
-    BigInt(finalAssetLimitWithTwoLastNumbersAsExponent) > 0n,
-    'expected finalAssetLimitWithManualSlippage to be a positive amount',
-  )
-
-  const aggregatorLastTwoChars = aggregator.slice(aggregator.length - 2, aggregator.length)
-  const finalAssetAddressLastTwoChars = finalAssetAddress.slice(
-    finalAssetAddress.length - 2,
-    finalAssetAddress.length,
-  )
   const shortenedPool = poolToShortenedPool[pool as keyof typeof poolToShortenedPool]
-
   assert(shortenedPool, 'cannot find shortened pool name')
 
-  // Thorchain memo format:
-  // SWAP:ASSET:DESTADDR:LIM:AFFILIATE:FEE:DEX Aggregator Addr:Final Asset Addr:MinAmountOut
-  // see https://gitlab.com/thorchain/thornode/-/merge_requests/2218 for reference
-  const memo = [
-    prefix,
-    shortenedPool,
-    address,
-    nativeAssetLimitWithManualSlippage,
-    affiliate,
-    affiliateBps,
-    aggregatorLastTwoChars,
-    finalAssetAddressLastTwoChars,
-    finalAssetLimitWithTwoLastNumbersAsExponent,
-  ].join(MEMO_PART_DELIMITER)
-
-  const memoBytesLength = new Blob([memo]).size
-
-  // UTXO only supports 80 bytes memo and we don't want to lose more precision
-  if (chainNamespace === CHAIN_NAMESPACE.Utxo) {
-    assert(memoBytesLength <= UTXO_MAXIMUM_BYTES_LENGTH, 'memo is too long')
+  const aggregatorLastTwoChars = aggregator.slice(-2)
+  const finalAssetAddressLastTwoChars = finalAssetAddress.slice(-2)
+
+  let exponent = 0
+  let formattedAmountOut = subtractBasisPointAmount(minAmountOut, slippageBps, BigNumber.ROUND_DOWN)
+
+  let potentialMemo
+  do {
+    let amountWithExponent = formattedAmountOut
+    if (exponent > 0) {
+      amountWithExponent = fromBaseUnit(amountWithExponent, exponent)
+      amountWithExponent = bn(amountWithExponent).integerValue(BigNumber.ROUND_DOWN).toFixed(0)
+    } else {
+      amountWithExponent = bn(formattedAmountOut).toFixed(0)
+    }
+    const amountOutStr =
+      amountWithExponent + (exponent === 0 ? '01' : exponent < 10 ? `0${exponent}` : `${exponent}`)
+
+    potentialMemo = [
+      prefix,
+      shortenedPool,
+      address,
+      nativeAssetLimitWithManualSlippage,
+      affiliate,
+      affiliateBps,
+      aggregatorLastTwoChars,
+      finalAssetAddressLastTwoChars,
```

Comment:
Be careful this part is not up to date, 2 assets is not enough to avoid colliding